### PR TITLE
Add blurb about CodeNarc tool autodetection to release notes

### LINKED ISSUE #21509: Default to latest version of CodeNarc tool
At time of writing, CodeNarc v3.1.0 is the most recent.

One complication - for Groovy 4 compatibility, the CodeNarc team published an alternate release using the same version number but different coordinates: 

The CodeNarc plugin should not assume that `localGroovy()` will be compatible with the version of the tool in use.

I've opened CodeNarc/CodeNarc#711 to ask the CodeNarc team to consider publishing Spock-style, where the bundled/expected Groovy runtime is indicated by the version string, not the artifact name.

### PR DESCRIPTION
**Pull request by tresat**
Fixes #21509 

See the "CodeNarc Plugin automatically detects appropriate version for current Groovy runtime" section in the [release notes](https://builds.gradle.org/repository/download/Gradle_Master_Check_BuildDistributions/57487108:id/distributions/gradle-8.0-docs.zip!/gradle-8.0-20221026125910%2B0000/docs/release-notes.html).

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test bd

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -75,10 +75,14 @@ ADD RELEASE FEATURES BELOW
...
+The [PMD](userguide/pmd_plugin.html) and [CodeNarc](userguide/codenarc_plugin.html) plugins now use the Gradle worker API and JVM toolchains. These tools now perform analysis via an external worker process and therefore their tasks may now run in parallel within one project.
 
 In Java projects, these tools will use the same version of Java required by the project. In other types of projects, they will use the same version of Java that is used by the Gradle daemon.
 
+#### CodeNarc Plugin automatically detects appropriate version for current Groovy runtime
+
+The [CodeNarc](https://codenarc.org/) project now publishes separate versions for use with Groovy 4.  Although Gradle still currently ships with Groovy 3, to ensure future compatibility the [CodeNarcPlugin](userguide/codenarc_plugin.html) will now by default automatically detect the appropriate version of CodeNarc to use for the current Groovy runtime.  The `toolVersion` property on the [CodeNarcExtension](current/dsl/org.gradle.api.plugins.quality.CodeNarcExtension.html#org.gradle.api.plugins.quality.CodeNarcExtension) can still be used to explicitly specify the version of CodeNarc to use.
```
Comment (nathan-contino) [inline subprojects/docs/src/docs/release/notes.md:84]:
```suggestion
The [CodeNarc](https://codenarc.org/) project now publishes separate versions for use with Groovy 4.
Gradle still currently ships with Groovy 3.
To ensure future compatibility, the [CodeNarcPlugin](userguide/codenarc_plugin.html) now automatically detects the appropriate version of CodeNarc for the current Groovy runtime.
You can still explicitly specify a CodeNarc version with the `toolVersion` property on the [CodeNarcExtension](current/dsl/org.gradle.api.plugins.quality.CodeNarcExtension.html#org.gradle.api.plugins.quality.CodeNarcExtension).
```
Suggestion: split up some long sentences, trim a few words.

**Review by nathan-contino — APPROVED** [review]
Docs LGTM! One suggestion.

File: subprojects/docs/src/docs/release/notes.md (hunk 1 above)
Comment (big-guy) [inline subprojects/docs/src/docs/release/notes.md:84]: This shouldn't be linking to /current/

**Review by big-guy — CHANGES_REQUESTED** [review]

**Comment by tresat** [comment]
@bot-gradle test this

**Review by big-guy — APPROVED** [review]

**Comment by tresat** [comment]
@bot-gradle test and merge

**Comment by tresat** [comment]
@bot-gradle test and merge

**Pull request merged**